applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_base_solver.py

```diff
@@ -132,63 +52,73 @@ def AddVariables(self):
+            mapper_project_parameters["interface_submodel_part_origin"].SetString(fluid_submodelpart_name)
+            mapper_project_parameters["interface_submodel_part_destination"].SetString(structure_submodelpart_name)
 
             self.interface_mapper = KratosMapping.MapperFactory.CreateMapper(self.fluid_solver.main_model_part,
-				                                                self.structure_solver.main_model_part,
-				                                                project_parameters_mapper)
+                                                                             self.structure_solver.main_model_part,
+                                                                             mapper_project_parameters)
 
             self.double_faced_structure = False
 
-        elif (self.settings["mapper_settings"].size() == 2):
+        elif (mapper_settings.size() == 2):
             # Get the fluid interface faces submodelpart names
             for mapper_id in range(2):
-                if (self.settings["mapper_settings"][mapper_id]["mapper_face"].GetString() == "Positive"):
-                    pos_face_submodelpart_name = self.settings["mapper_settings"][mapper_id]["fluid_interface_submodelpart_name"].GetString()
-                elif (self.settings["mapper_settings"][mapper_id]["mapper_face"].GetString() == "Negative"):
-                    neg_face_submodelpart_name = self.settings["mapper_settings"][mapper_id]["fluid_interface_submodelpart_name"].GetString()
+                if (mapper_settings[mapper_id]["mapper_face"].GetString() == "Positive"):
+                    pos_face_submodelpart_name = mapper_settings[mapper_id]["fluid_interface_submodelpart_name"].GetString()
+                elif (mapper_settings[mapper_id]["mapper_face"].GetString() == "Negative"):
+                    neg_face_submodelpart_name = mapper_settings[mapper_id]["fluid_interface_submodelpart_name"].GetString()
                 else:
                     raise Exception("Unique mapper flag has been set but more than one mapper exist in mapper_settings.")
             # Get the structure submodelpart name
-            structure_submodelpart_name = self.settings["mapper_settings"][0]["structure_interface_submodelpart_name"].GetString()
+            structure_submodelpart_name = mapper_settings[0]["structure_interface_submodelpart_name"].GetString()
 
             # Grab the interface submodelparts
             pos_fluid_submodelpart = self.fluid_solver.main_model_part.GetSubModelPart(pos_face_submodelpart_name)
             neg_fluid_submodelpart = self.fluid_solver.main_model_part.GetSubModelPart(neg_face_submodelpart_name)
             structure_submodelpart = self.structure_solver.main_model_part.GetSubModelPart(structure_submodelpart_name)
 
-            # TODO: SET THE DOUBLE SIDE SURFACE MAPPING
+            # TODO: SET THE DOUBLE SIDE SURFACE 
+            raise Exception ("_SetUpMapper not implemented yet for double side surfaces")
```

Comment:
side remark: this should be quite simple, you only need the `ADD_VALUES` argument for the mapping functions of the 2-n mapper to not overwrite the previous values

Reply:
Done! Thanks @philbucher !